### PR TITLE
better monitoring of convox start

### PR DESCRIPTION
**Pull request by awsmsrc**
## Release Playbook
- [ ] Rebase against master
- [ ] Release branch ()
- [ ] Pass CI ()
- [ ] Code review
- [ ] Merge into master
- [ ] Release master ()
- [ ] Pass CI ()
- [ ] Update staging rack
- [ ] Edit [Rack release record](https://github.com/convox/rack/releases) and/or update [docs](https://github.com/convox/convox.github.io)
- [ ] Publish release
- [ ] Release CLI

### CODE REVIEW

**Comment by ddollar** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;6u16hHT8&#x2F;297-convox-start-reports-application-type"> convox start reports application type</a>

File: cmd/convox/start.go [hunk 1]
```diff
@@ -64,31 +73,33 @@ func cmdStart(c *cli.Context) error {
 	}
 
 	if pcc, err := m.PortConflicts(); err != nil || len(pcc) > 0 {
-		if err != nil {
-			stdcli.ExitError(err)
+		if err == nil {
+			err = fmt.Errorf("ports in use: %v", pcc)
```
Comment (ddollar) [inline cmd/convox/start.go]: This seems like it disables the error

File: cmd/convox/start.go [hunk 2]
```diff
@@ -64,31 +73,33 @@ func cmdStart(c *cli.Context) error {
...
+	})
 
-	go handleInterrupt(r)
+	go handleInterrupt(r, id)
 
 	return r.Wait()
 }
 
-func handleInterrupt(run manifest.Run) {
+func handleInterrupt(run manifest.Run, id string) {
```
Comment (ddollar) [inline cmd/convox/start.go]: Why the parameter list change?

File: cmd/convox/stdcli/stdcli.go [hunk 3]
```diff
@@ -211,7 +221,7 @@ func QOSEventSend(system, id string, ep QOSEventProperties) error {
 
 	rollbar.Wait()
 
-	if ep.Error != nil {
+	if ep.Error != nil || ep.ValidationError != nil {
 		return ExitError(ep.Error)
```
Comment (ddollar) [inline cmd/convox/stdcli/stdcli.go]: This displays ep.Error no matter which one was not nil.

File: cmd/convox/start.go (hunk 1 above)
Comment (awsmsrc) [inline cmd/convox/start.go]: if the error is nil then and there ARE port conflicts we create a new validation error and exit with that
if the error is not nil then something went wrong parsing the manifest and we error out with that instead

QOSEventSend exits out if there was an error

File: cmd/convox/stdcli/stdcli.go (hunk 3 above)
Comment (awsmsrc) [inline cmd/convox/stdcli/stdcli.go]: You are correct sir!

**Comment by awsmsrc** [comment]
gonna merge this into start-refactor and handle as one change please re review post humously though @MiguelMoll @nzoschke

**Pull request merged**